src/Storages/Distributed/DistributedBlockOutputStream.cpp

```diff
@@ -622,6 +628,13 @@ void DistributedBlockOutputStream::writeToShard(const Block & block, const std::
 
         /// Write batch to temporary location
         {
+            std::optional<DirectorySyncGuard> tmp_path_sync_guard;
```

Comment:
will it be recreated on every iteration?

Reply:
There is no loop, there is just a sub scope (for objects lifetime)